[DataGrid] Include MUI core component localizations in `localeText` (#1913)
mui-org/material-ui-x: docs/src/pages/components/data-grid/localization/localization.md

Note that `createMuiTheme` accepts any number of arguments.
If you are already using the [translations of the core components](/guides/localization/#locale-text), you can add `bgBG` as a new argument.
The same import works for `XGrid` as it's an extension of `DataGrid`.

```jsx
import { createMuiTheme, ThemeProvider } from '@material-ui/core/styles';
import { DataGrid, bgBG } from '@material-ui/data-grid';
import { bgBG as coreBgBG } from '@material-ui/core/locale';

const theme = createMuiTheme(
  {
    palette: {
      primary: { main: '#1976d2' },
    },
  },
  bgBG,
  coreBgBG,

> );

Reviewer: 👍

> If you are already using the [translations of the core components](/guides/localization/#locale-text), you can add `bgBG` as a new argument.

Reviewer: If the user is already using the translations from core, using this approach will break the core components, because we don't append anymore the core translation in our translations. We need to explain that `localeText` has to be used. Also, could you update the PR's description to include a migration path?

Author: Am no, why you can add both the translations from the `DataGrid` along with the translations from the core. You can do :
```
const theme = createMuiTheme(
  {
    palette: {
      primary: { main: '#1976d2' },
    },
  },
  CoreBgBG,
  DataGridBgBG,
);
```
Regarding the migration, one would just need to add the translations from the core as an additional argument to the `createMuiTheme`.

Reviewer 2: @DanailH 👍 for adding this as an example.

Author: Added

> The same import works with `XGrid` as it's an extension of `DataGrid`.

Reviewer: Just to avoid breaking the line in the the period sign.

<img width="200" src="https://user-images.githubusercontent.com/42154031/122558975-17f8bd00-d015-11eb-9f11-b989ddffb1be.png" alt="image" style="max-width:100%;">

```suggestion
The same import works for `XGrid` as it's an extension of `DataGrid`.
```

> **Note**: if you want to pass in language translations directly to the grid and not use the `createMuiTheme` in combination with the `ThemeProvider` you can directly load the language translations from the `DataGrid` or `XGrid` package.

Reviewer: ```suggestion
If you want to pass language translations directly to the grid without using `createMuiTheme` and `ThemeProvider`, you can directly load the language translations from the `@material-ui/data-grid` or `@material-ui/x-grid` package.
```

> **Note**: if you want to pass in language translations directly to the grid and not use the `createMuiTheme` in combination with the `ThemeProvider` you can directly load the language translations you need from the `DataGrid` or `XGrid` package.

Reviewer 3: I would remove `you need`
